UACloudLibraryServer/Startup.cs

```diff
@@ -73,8 +73,10 @@ public void ConfigureServices(IServiceCollection services)
                 o.UseNpgsql(PostgreSQLDB.CreateConnectionString());
             });
 
-            services.AddDefaultIdentity<IdentityUser>(options => options.SignIn.RequireConfirmedAccount = true)
-                .AddEntityFrameworkStores<AppDbContext>();
+            services.AddDefaultIdentity<IdentityUser>(options => 
+                    //require confirmation mail if sendgrid API Key is set
+                    options.SignIn.RequireConfirmedAccount = !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("SendGridAPIKey"))

```

Comment:
Environment.GetEnvironmentVariable() doesn't work in startup.cs, use Configuration[] instead.